dlc-oracle/src/main/scala/org/bitcoins/dlc/oracle/storage/SeedStorage.scala

```diff
@@ -0,0 +1,243 @@
+        Left(ReadMnemonicError.NotFoundError)
+      }
+    }
+
+    import MnemonicJsonKeys._
+    import ReadMnemonicError._
+
+    val readJsonTupleEither: Either[
+      ReadMnemonicError,
+      (String, String, String, Long)] = jsonE.flatMap { json =>
+      logger.trace(s"Read encrypted mnemonic JSON: $json")
+      val creationTimeNum = Try(json(CREATION_TIME).num.toLong) match {
+        case Success(value) =>
+          value
+        case Failure(err) if err.isInstanceOf[NoSuchElementException] =>
+          FIRST_WALLET_TIME
+        case Failure(exception) => throw exception
+      }
+      Try {
+        val ivString = json(IV).str
+        val cipherTextString = json(CIPHER_TEXT).str
+        val rawSaltString = json(SALT).str
+        (ivString, cipherTextString, rawSaltString, creationTimeNum)
+      } match {
+        case Success(value)     => Right(value)
+        case Failure(exception) => throw exception
+      }
+    }
+
+    val encryptedEither: Either[ReadMnemonicError, EncryptedMnemonic] =
+      readJsonTupleEither.flatMap {
+        case (rawIv, rawCipherText, rawSalt, rawCreationTime) =>
+          val encryptedOpt = for {
+            iv <- ByteVector.fromHex(rawIv).map(AesIV.fromValidBytes)
+            cipherText <- ByteVector.fromHex(rawCipherText)
+            salt <- ByteVector.fromHex(rawSalt).map(AesSalt(_))
+          } yield {
+            logger.debug(
+              s"Parsed contents of $seedPath into an EncryptedMnemonic")
+            EncryptedMnemonic(AesEncryptedData(cipherText, iv),
+                              salt,
+                              Instant.ofEpochSecond(rawCreationTime))
+          }
+          val toRight: Option[Right[ReadMnemonicError, EncryptedMnemonic]] =
+            encryptedOpt
+              .map(Right(_))
+
+          toRight.getOrElse(
+            Left(JsonParsingError("JSON contents was not hex strings")))
+      }
+    encryptedEither
+  }
+
+  /**
+    * Reads the wallet mnemonic from disk and tries to parse and
+    * decrypt it
+    */
+  def decryptMnemonicFromDisk(
+      seedPath: Path,
+      passphrase: AesPassword): DecryptedMnemonic = {
+
+    val encryptedEither = readEncryptedMnemonicFromDisk(seedPath)
+
+    val decryptedEither: Either[ReadMnemonicError, DecryptedMnemonic] =
+      encryptedEither.flatMap { encrypted =>
+        encrypted.toMnemonic(passphrase) match {
+          case Failure(exc) =>
+            logger.error(s"Error when decrypting $encrypted: $exc")
+            Left(ReadMnemonicError.DecryptionError)
+          case Success(mnemonic) =>
+            logger.debug(s"Decrypted $encrypted successfully")
+            val decryptedMnemonic =
+              DecryptedMnemonic(mnemonic, encrypted.creationTime)
+            Right(decryptedMnemonic)
+        }
+      }
+
+    decryptedEither match {
+      case Left(err)    => sys.error(err.toString)
+      case Right(value) => value
+    }
+  }
+
+  def getPrivateKeyFromDisk(
```

Comment:
is there any way we can limit scope on this ? Although I guess we have the same problem in the key manager project currently with `WalletStorage.decryptMnemonicFromDisk()`. In fact, can't you just re-use the code since we expose that as a public api currently?